### PR TITLE
CLOUDSTACK-8968: UI icon over VM snapshot to deploy user instance

### PR DESCRIPTION
**Pull request by nitin-maharana**
Added a new Icon in Instance page to launch the VM from the snapshot.

A new icon over VM snapshot object, which upon invoked will open a deploy user instance wizard, where there is no choice to choose zone and ISO or template to deploy the user instance from.

1) A new icon to indicate deploy user instance from VM snapshot
2) The new icon is placed as an operation over each VM snapshot object.
3) Clicking this icon will show the wizard to deploy user instance, which is same as wizard for normal user instance except that zone selection page and ISO/template selection page would not be present in this wizard.

### CODE REVIEW

File: ui/scripts/instanceWizard.js [hunk 1]
```diff
@@ -731,25 +746,33 @@
...
 
-            if (args.$wizard.find('input[name=rootDiskSize]').parent().css('display') != 'none')  {
-                if (args.$wizard.find('input[name=rootDiskSize]').val().length > 0) {
-                    $.extend(deployVmData, {
-                        rootdisksize : args.$wizard.find('input[name=rootDiskSize]').val()
-                    });
-                }
-            }
+//            Currently it is not supporting in backend
+//            if (args.$wizard.find('input[name=rootDiskSize]').parent().css('display') != 'none')  {
```
Comment (DaanHoogland) [inline ui/scripts/instanceWizard.js]: please do not leave code commented in. this block should be deleted.

**Comment by bhaisaab** [comment]
@nitin-maharana if you can address what Daan has commented today, along with fix tabs with spaces (if any) I can help review this and merge this; before 4.7.0/master freeze is announced today.

**Comment by remibergsma** [comment]
@nitin-maharana @bhaisaab I looked around but just don't find what's changed. Please post screenshots of what is changed or instruct exactly how it should work. Thanks!

**Comment by remibergsma** [comment]
Hmm.. this isn't supported on KVM, is it? That may be why I can't find it.

**Comment by nitin-maharana** [comment]
@bhaisaab @DaanHoogland I updated the change with current 4.6 branch and removed the commented codes. @remibergsma This only supports in VMware. Thanks :)

**Comment by DaanHoogland** [comment]
@nitin-maharana @bhaisaab both I and @remibergsma don't work with vmware as part of our cloudstack installs, can you do testing and preferably write tests for proving the continued funtioning of the change?

**Comment by DaanHoogland** [comment]
@bhaisaab did you test this? should we merge before the RC tonight?

**Comment by bhaisaab** [comment]
@DaanHoogland sorry missed your message, I did not test this

**Comment by nitin-maharana** [comment]
Step-1:
=====
pr_1150_ss1_nitin

Step-2:
=====
Skipped 1st and 2nd steps

pr_1150_ss2_nitin

Step-3:
=====
Last step

pr_1150_ss3_nitin

**Comment by nitin-maharana** [comment]
cc @bhaisaab

**Comment by bhaisaab** [comment]
@nitin-maharana sorry don't have the infra to setup vmware and test right now, so cannot help test it

**Comment by nitin-maharana** [comment]
@remibergsma : Any updates on this? Can you please look at the change if you have a setup. Thanks.

**Comment by remibergsma** [comment]
Sorry I have no vmware setup to test this.

**Comment by nitin-maharana** [comment]
Closing this PR as made a new PR #1320 (Against 4.7 which will be merged in master later).

**Pull request closed without merging**